### PR TITLE
added Ruby code generator

### PR DESCRIPTION
**Pull request by StrikerRUS**
Ruby is in top15 TIOBE latest rating: https://www.tiobe.com/tiobe-index/. 🙂 

image

I expect that adding Ruby support will be demanded not only in pure Ruby itself, but in **Ruby on Rails** web framework which is among popular frameworks: [1](https://medium.com/front-end-weekly/10-most-popular-web-frameworks-in-2020-167b9103e08a), [2](https://hotframeworks.com/).

### CODE REVIEW

File: m2cgen/interpreters/ruby/interpreter.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        if self.with_linear_algebra:
+            filename = os.path.join(
+                os.path.dirname(__file__), "linear_algebra.rb")
+            self._cg.prepend_code_lines(utils.get_file_content(filename))
+
+        return self._cg.code
+
+    def interpret_bin_num_expr(self, expr, **kwargs):
+        if expr.op == ast.BinNumOpType.DIV:
+            # Always force float result
```
Comment (izeigerman) [inline m2cgen/interpreters/ruby/interpreter.py:49]: A brief explanation why would be nice

File: m2cgen/interpreters/ruby/code_generator.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        self.increase_indent()
+
+    @contextlib.contextmanager
+    def function_definition(self, name, args):
+        self.add_function_def(name, args)
+        yield
+        self.add_block_termination()
+
+    def method_invocation(self, method_name, obj, args):
+        return ("(" + str(obj) + ")." + method_name +
```
Comment (izeigerman) [inline m2cgen/interpreters/ruby/code_generator.py:33]: Why do we need these parentheses (the ones in code not in string)? It makes the code a bit harder to digest

Comment (izeigerman) [reply m2cgen/interpreters/ruby/code_generator.py:33]: Btw, while I'm on it, why do we need parentheses around the object?

File: m2cgen/interpreters/ruby/interpreter.py (hunk 1 above)
Comment (StrikerRUS) [reply m2cgen/interpreters/ruby/interpreter.py:49]: image

https://stackoverflow.com/questions/5502761/why-is-division-in-ruby-returning-an-integer-instead-of-decimal-value

File: m2cgen/interpreters/ruby/code_generator.py (hunk 2 above)
Comment (StrikerRUS) [reply m2cgen/interpreters/ruby/code_generator.py:33]: It was my initial mistake to not use these parentheses 😃 . It was cough by one of our e2e tests. Then I realized that
`1.0 + 2.0.fdiv(3)` is not the same as `(1.0 + 2.0).fdiv(3)`.

File: m2cgen/interpreters/ruby/interpreter.py (hunk 1 above)
Comment (StrikerRUS) [reply m2cgen/interpreters/ruby/interpreter.py:49]: I find using `fdiv` less polluting code than explicit converting every number to float with `.to_f` method.

File: m2cgen/interpreters/ruby/code_generator.py (hunk 2 above)
Comment (izeigerman) [reply m2cgen/interpreters/ruby/code_generator.py:33]: ah, makes perfect sense 👍

**Pull request merged**